src/frequenz/sdk/timeseries/_base_types.py

```diff
@@ -34,6 +34,10 @@ class Sample(Generic[QuantityT]):
     value: QuantityT | None = None
     """The value of this sample."""
 
+    def __str__(self) -> str:
+        """Return a string representation of the sample."""
+        return f"Sample({str(self.timestamp)}, {str(self.value)})"
```

Comment:
Why calling `str` explicitly? This is what should be done when you interpolate anyways. Also it's better to use `type(self).__name__` in case someone subclasses it and doesn't override the `str`. Finally, it would be nice to have a `repr` too:

```suggestion
    def __str__(self) -> str:
        """Return a string representation of the sample."""
        return f"{type(self).__name__}({self.timestamp}, {self.value})"

    def __repr__(self) -> str:
        """Return a string representation of the sample."""
        return f"{type(self).__name__}({self.timestamp=}, {self.value=})"
```